### PR TITLE
[Fix] Fix deepseek training 2

### PR DESCRIPTION
**Pull request by HAOCHENYE**

### CODE REVIEW

File: xtuner/v1/module/router/noaux_router.py [hunk 1]
```diff
@@ -71,6 +71,38 @@ def forward(self, logits) -> RouterResults:
             noise = torch.randn_like(scores) * 50
             scores_for_choice = scores + noise
 
+        if self.n_group != self.topk_group:
+            assert len(logits.shape) == 2, (
+                f"XTuner Interval bug, invalid logits shape: {logits.shape}, expected shape with "
```
Comment (pppppM) [inline xtuner/v1/module/router/noaux_router.py:76]: internal?

Comment (HAOCHENYE) [reply xtuner/v1/module/router/noaux_router.py:76]: Since `logits` is computed by the gate with output shape `(seq_len × batch_size, hidden_states)`, I raise an `Internal Error` if `logits` is not 2D, indicating the router here was called unexpectedly.

File: xtuner/v1/model/moe/moe.py [hunk 2]
```diff
@@ -375,12 +376,14 @@ def _micro_batch_forward(
                         seq_ctx=seq_ctx_list,
                     )
                 hidden_states = layer_results[: len(hidden_states_list)]
-                router_logits = layer_results[len(hidden_states_list) :]
+                router_logits = layer_results[len(hidden_states_list) : len(hidden_states_list) * 2]
```
Comment (hhaAndroid) [inline xtuner/v1/model/moe/moe.py:379]: qwen3vl_text.py 也要对应修改
